v2/RNHealthKitCore/HealthKitCore/Workout/WorkoutQueries.swift

```diff
@@ -35,25 +35,51 @@ public extension HealthKitCore {
     ///   - endDate: The end date and time of the workout.
     ///   - totalEnergyBurned: The total energy burned during the workout (in kilocalories).
     ///   - totalDistance: The total distance covered during the workout (in kilometers).
+    ///   - metadata: Optional metadata for the workout.
     /// - Throws: An error if the workout data cannot be saved to HealthKit.
     func saveCompletedWorkout(
         activityType: HKWorkoutActivityType,
         startDate: Date,
         endDate: Date,
         totalEnergyBurned: Double?,
-        totalDistance: Double?
-    ) async throws {
-        let workout = HKWorkout(
```

Comment:
Apple is deprecating `HKWorkout` initializer on iOS 17. I replaced with `HKWorkoutBuilder` which has support from iOS 12+.

Reply:
cool